lib/src/dio_http_formatter_base.dart

```diff
@@ -104,7 +104,7 @@ class HttpFormatter extends Interceptor {
         }
       }
 
-      if (_includeRequestBody && requestOptions.data != null) {
+      if (_includeRequestBody && !(requestOptions.data?.isEmpty ?? true)) {
```

Comment:
@JleMyP It's preferable that the condition be more explicit and readable, like:
```suggestion
      if (_includeRequestBody && requestOptions.data != null && requestOptions.data.isNotEmpty) {
```